### DISCUSSION TITLE
Modular W-DS

### DISCUSSION THREAD

**Discussion by NightScript370** [Epicpkmn11/WordleDS#11] [Support and Suggestions]
After lifehackerhansle introduced me to Jazle (A clone of Web Wordle with JavaScript library words), I've became increasingly interested in porting all sorts of Wordle clones to the Nintendo DS. Currently, for Jazle DS, I am injecting my word list into the code and then compiling a new build, which isn't very helpful for the Wordle DS ecosystem. I've also attempted custom how-to-play text, but that already broke its graphics.

Switching to reading files from the SD card would be much more elegant, especially considering that when there is a bug in the main Wordle-DS engine, an update to it wouldn't mean requiring to update all its forks.

Said files from the SD card should be INI files for its configuration, and JSON files for the lists.

Also, instead of hardcoding the amount of letters each series could have, we should have it look for the amount of letters in the word.

New dependencies it would have to introduce would be three, but we are already using external dependencies so it couldn't really hurt.

Opinions are welcome.

**Comment by Epicpkmn11** [comment]
Initial mod support is added in d1d7979d4b95758bce2c079fd16c647148042415, it still needs a bit of polish at the moment but it should be functional. Haven't updated the README/wiki yet but it's not too complicated:
- Mods go in `sd:/_nds/WordleDS/[mod]` (config files and such are also moved here now)
- inside the mod folder (can be any name) you can have several files to customize the game, all optional:
   - `mod.json` controls the word lists, emoji to write, win messages, keyboard layout, etc (anything left out of this file will use the default, anything invalid should throw an assert explaining what's wrong on load) ([examples](https://github.com/Epicpkmn11/WordleDS/tree/main/resources/mod%20examples))
   - `music.msl`, same as before, if you had custom music you now need to move it here
   - All [images](https://github.com/Epicpkmn11/WordleDS/tree/main/gfx) can be replaced using GRF files with the same names and conversion settings as they are in the repo, I might try throw together some web app for easier GRF conversion
   - [Fonts](https://github.com/Epicpkmn11/WordleDS/tree/main/data) are the same names as in the repo
- Then (for now) open `sd:/_nds/WordleDS/settings.json` (run the new version once to make this) in a text editor and change the `"mod"` to the name of your mod folder, I'll add a way to change in settings tomorrow
- Stats are stored individually for each mod at `sd:/_nds/WordleDS/mod/stats.json`

**Reply by Epicpkmn11** [reply]
With df7396c0f400a5cd079f5f0167bebb1dfe9473d0 mod support is nearly done, started working on a [wiki page](https://github.com/Epicpkmn11/WordleDS/wiki/modding) explaining it but that's still WIP.

Probably still a few bugs to iron out but it should be mostly stable.

**Reply by Epicpkmn11** [reply]
Mod support is more or less complete now, I've opened up another discussion and a branch for sharing mods and more or less finished up the wiki. I'll probably do a new release later today.
